com.minres.coredsl/src/com/minres/coredsl/analysis/CoreDslAnalyzer.xtend

```diff
@@ -0,0 +1,217 @@
+package com.minres.coredsl.analysis
+
+import com.minres.coredsl.coreDsl.AssignmentExpression
+import com.minres.coredsl.coreDsl.Attribute
+import com.minres.coredsl.coreDsl.CoreDef
+import com.minres.coredsl.coreDsl.CoreDslPackage
+import com.minres.coredsl.coreDsl.Declaration
+import com.minres.coredsl.coreDsl.DeclarationStatement
+import com.minres.coredsl.coreDsl.Declarator
+import com.minres.coredsl.coreDsl.DescriptionContent
+import com.minres.coredsl.coreDsl.EntityReference
+import com.minres.coredsl.coreDsl.Expression
+import com.minres.coredsl.coreDsl.ExpressionStatement
+import com.minres.coredsl.coreDsl.ISA
+import com.minres.coredsl.coreDsl.InstructionSet
+import com.minres.coredsl.coreDsl.IntegerTypeSpecifier
+import com.minres.coredsl.coreDsl.StorageClassSpecifier
+import com.minres.coredsl.coreDsl.TypeQualifier
+import com.minres.coredsl.coreDsl.TypeSpecifier
+import com.minres.coredsl.validation.IssueCodes
+import com.minres.coredsl.validation.KnownAttributes.AttributeUsage
+import java.util.HashSet
+import org.eclipse.xtext.validation.ValidationMessageAcceptor
+
+import static extension com.minres.coredsl.util.ModelExtensions.*
+import com.minres.coredsl.coreDsl.BoolTypeSpecifier
+import com.minres.coredsl.coreDsl.VoidTypeSpecifier
+
+class CoreDslAnalyzer {
+
+	def static void analyze(DescriptionContent desc, ValidationMessageAcceptor acceptor) {
+		val ctx = new AnalysisContext(acceptor);
+
+		for (isa : desc.definitions) {
+			analyzeIsaArchState(ctx, isa);
+			val ectx = CoreDslElaborator.elaborate(ctx, isa);
+			ctx.elaborationResults.put(isa, ectx);
+		}
+	}
+
+	def private static dispatch void analyzeIsaArchState(AnalysisContext ctx, InstructionSet iset) {
+		if(ctx.exposedDeclarations.containsKey(iset)) return;
+
+		val declaredNames = new HashSet<String>();
```

Comment:
Similar as above: Please check if a `LinkedHashSet` would be desirable here.

Reply:
Not needed, this set is only used to check whether a name has already been declared.